ecr-login/api/factory.go

```diff
@@ -65,6 +74,7 @@ func (defaultClientFactory DefaultClientFactory) NewClient(awsSession *session.S
 
 // NewClientWithOptions Create new client with Options
 func (defaultClientFactory DefaultClientFactory) NewClientWithOptions(opts Options) Client {
+	opts.Session.Handlers.Build.PushBackNamed(userAgentHandler)
```

Comment:
I think you can remove it from here.  `NewClientWithOptions` is only ever called by `NewClientWithDefaults` and `NewClientFromRegion` in this program, so it'll already have the handler.  We still have this function because it can be useful in [other programs](https://github.com/aws/amazon-ecs-cli/blob/0bfcecd20eb96e5d4293b4dde9839d6ea6e5bdea/ecs-cli/modules/clients/aws/ecr/client.go#L60) which may already be setting their own user-agent.

Reply:
This has been fixed by 33a1bc7